pydantic/_internal/_generate_schema.py

```diff
@@ -94,8 +94,7 @@ def check_validator_fields_against_field_name(
     info: FieldDecoratorInfo,
     field: str,
 ) -> bool:
-    if isinstance(info, ValidatorDecoratorInfo):
-        # V1 compat: accept `'*'` as a wildcard that matches all fields
+    if isinstance(info, (ValidatorDecoratorInfo, FieldValidatorDecoratorInfo)):
         if info.fields == ('*',):
```

Comment:
I guess maybe this is where it should be changed to allow `'*'` to be present, not require exact equality on the fields (i.e., not allow others).

Reply:
Yeah looks like on this branch, I get this error when I attempt to run the above:
```
pydantic.errors.PydanticUserError: Validators defined with incorrect fields: __main__.Model:5133041408.validate (use check_fields=False if you're inheriting from the model and intended this)
```
I think the code above should allow this. (Or if not, it should at least give a clearer error message. But I think it should be allowed.)

Reply:
This is a bit weird... And I actually didn't know people use it, but I've applied it. 🙏